search/elastic.py

```diff
@@ -579,35 +558,34 @@ def search(self,
         if elastic_filters:
             filter_segment = {
                 "bool": {
-                    "must": elastic_filters
+                    "should": elastic_filters
                 }
             }
             query = {
-                "filtered": {
-                    "query": query_segment,
+                "bool": {
+                    "must": query_segment,
                     "filter": filter_segment,
                 }
             }
 
-        body = {"query": query}
-        if facet_terms:
-            facet_query = _process_facet_terms(facet_terms)
-            if facet_query:
-                body["facets"] = facet_query
+        if exclude_dictionary:
+            if query.get("bool"):
+                query["bool"]["must_not"] = _process_exclude_dictionary(exclude_dictionary)
+            else:
+                query = {
+                    "bool": {
+                        "must_not": _process_exclude_dictionary(exclude_dictionary)
+                    }
+                }
+
+        body = {"query": query, }
```

Comment:
Comma again. Come on, that is a single line (not multi), not tuple...

Reply:
Done